### ISSUE TITLE
Cannot read webcam on macOS

### ISSUE THREAD

**Issue by florianletsch** [imageio/imageio#391]
Following the [webcam example in the docs](https://imageio.readthedocs.io/en/stable/examples.html#grab-frames-from-your-webcam), I am trying to access the webcam on my MacBook (external Logitech camera attached).

```
import imageio
reader = imageio.get_reader('<video0>')

# ... long error message (see full listing below)
# IndexError: No video4linux camera at <video0>.
```

This call fails with quite an extensive error message (see below)

ffmpeg 4.0.2 is installed from homebrew (`brew install ffmpeg`).

```
ffmpeg
ffmpeg version 4.0.2 Copyright (c) 2000-2018 the FFmpeg developers
  built with Apple LLVM version 10.0.0 (clang-1000.10.43.1)
  configuration: --prefix=/usr/local/Cellar/ffmpeg/4.0.2 --enable-shared --enable-pthreads --enable-version3 --enable-hardcoded-tables --enable-avresample --cc=clang --host-cflags= --host-ldflags= --enable-gpl --enable-libmp3lame --enable-libx264 --enable-libxvid --enable-opencl --enable-videotoolbox --disable-lzma
  libavutil      56. 14.100 / 56. 14.100
  libavcodec     58. 18.100 / 58. 18.100
  libavformat    58. 12.100 / 58. 12.100
  libavdevice    58.  3.100 / 58.  3.100
  libavfilter     7. 16.100 /  7. 16.100
  libavresample   4.  0.  0 /  4.  0.  0
  libswscale      5.  1.100 /  5.  1.100
  libswresample   3.  1.100 /  3.  1.100
  libpostproc    55.  1.100 / 55.  1.100
Hyper fast Audio and Video encoder
usage: ffmpeg [options] [[infile options] -i infile]... {[outfile options] outfile}...

Use -h to get full help or, even better, run 'man ffmpeg'
```

Is this related to ffmpeg or imageio?

How can I check if ffmpeg itself is able to read the webcam properly?

Error message:

```
---------------------------------------------------------------------------
IndexError                                Traceback (most recent call last)
<ipython-input-3-71e67521605e> in <module>
----> 1 reader = imageio.get_reader('<video0>')

~/miniconda3/envs/test-imageio/lib/python3.6/site-packages/imageio/core/functions.py in get_reader(uri, format, mode, **kwargs)
    141
    142     # Return its reader object
--> 143     return format.get_reader(request)
    144
    145

~/miniconda3/envs/test-imageio/lib/python3.6/site-packages/imageio/core/format.py in get_reader(self, request)
    172                 "Format %s cannot read in mode %r" % (self.name, select_mode)
    173             )
--> 174         return self.Reader(self, request)
    175
    176     def get_writer(self, request):

~/miniconda3/envs/test-imageio/lib/python3.6/site-packages/imageio/core/format.py in __init__(self, format, request)
    222             self._request = request
    223             # Open the reader/writer
--> 224             self._open(**self.request.kwargs.copy())
    225
    226         @property

~/miniconda3/envs/test-imageio/lib/python3.6/site-packages/imageio/plugins/ffmpeg.py in _open(self, loop, size, dtype, pixelformat, print_info, ffmpeg_params, input_params, output_params, fps)
    397             # Start ffmpeg subprocess and get meta information
    398             self._initialize()
--> 399             self._load_infos()
    400
    401             # For cameras, create thread that keeps reading the images

~/miniconda3/envs/test-imageio/lib/python3.6/site-packages/imageio/plugins/ffmpeg.py in _load_infos(self)
    572                     raise IndexError(
    573                         "No video4linux camera at %s.\n\n%s"
--> 574                         % (self.request._video, ffmpeg_err)
    575                     )
    576                 else:

IndexError: No video4linux camera at <video0>.

FFMPEG STDERR OUTPUT:

ffmpeg version 4.0.2 Copyright (c) 2000-2018 the FFmpeg developers
  built with Apple LLVM version 10.0.0 (clang-1000.10.43.1)
  configuration: --prefix=/usr/local/Cellar/ffmpeg/4.0.2 --enable-shared --enable-pthreads --enable-version3 --enable-hardcoded-tables --enable-avresample --cc=clang --host-cflags= --host-ldflags= --enable-gpl --enable-libmp3lame --enable-libx264 --enable-libxvid --enable-opencl --enable-videotoolbox --disable-lzma
  libavutil      56. 14.100 / 56. 14.100
  libavcodec     58. 18.100 / 58. 18.100
  libavformat    58. 12.100 / 58. 12.100
  libavdevice    58.  3.100 / 58.  3.100
  libavfilter     7. 16.100 /  7. 16.100
  libavresample   4.  0.  0 /  4.  0.  0
  libswscale      5.  1.100 /  5.  1.100
  libswresample   3.  1.100 /  3.  1.100
  libpostproc    55.  1.100 / 55.  1.100
[avfoundation @ 0x7fc19c002000] Selected framerate (29.970030) is not supported by the device
[avfoundation @ 0x7fc19c002000] Supported modes:
[avfoundation @ 0x7fc19c002000]   160x90@[30.000030 30.000030]fps
[avfoundation @ 0x7fc19c002000]   160x90@[24.000038 24.000038]fps
[avfoundation @ 0x7fc19c002000]   160x90@[20.000000 20.000000]fps
[avfoundation @ 0x7fc19c002000]   160x90@[15.000015 15.000015]fps
[avfoundation @ 0x7fc19c002000]   160x90@[10.000000 10.000000]fps
[avfoundation @ 0x7fc19c002000]   160x90@[7.500002 7.500002]fps
[avfoundation @ 0x7fc19c002000]   160x90@[5.000000 5.000000]fps
[avfoundation @ 0x7fc19c002000]   160x120@[30.000030 30.000030]fps
[avfoundation @ 0x7fc19c002000]   160x120@[24.000038 24.000038]fps
[avfoundation @ 0x7fc19c002000]   160x120@[20.000000 20.000000]fps
[avfoundation @ 0x7fc19c002000]   160x120@[15.000015 15.000015]fps
[avfoundation @ 0x7fc19c002000]   160x120@[10.000000 10.000000]fps
[avfoundation @ 0x7fc19c002000]   160x120@[7.500002 7.500002]fps
[avfoundation @ 0x7fc19c002000]   160x120@[5.000000 5.000000]fps
[avfoundation @ 0x7fc19c002000]   176x144@[30.000030 30.000030]fps
[avfoundation @ 0x7fc19c002000]   176x144@[24.000038 24.000038]fps
[avfoundation @ 0x7fc19c002000]   176x144@[20.000000 20.000000]fps
[avfoundation @ 0x7fc19c002000]   176x144@[15.000015 15.000015]fps
[avfoundation @ 0x7fc19c002000]   176x144@[10.000000 10.000000]fps
[avfoundation @ 0x7fc19c002000]   176x144@[7.500002 7.500002]fps
[avfoundation @ 0x7fc19c002000]   176x144@[5.000000 5.000000]fps
[avfoundation @ 0x7fc19c002000]   320x180@[30.000030 30.000030]fps
[avfoundation @ 0x7fc19c002000]   320x180@[24.000038 24.000038]fps
[avfoundation @ 0x7fc19c002000]   320x180@[20.000000 20.000000]fps
[avfoundation @ 0x7fc19c002000]   320x180@[15.000015 15.000015]fps
[avfoundation @ 0x7fc19c002000]   320x180@[10.000000 10.000000]fps
[avfoundation @ 0x7fc19c002000]   320x180@[7.500002 7.500002]fps
[avfoundation @ 0x7fc19c002000]   320x180@[5.000000 5.000000]fps
[avfoundation @ 0x7fc19c002000]   320x240@[30.000030 30.000030]fps
[avfoundation @ 0x7fc19c002000]   320x240@[24.000038 24.000038]fps
[avfoundation @ 0x7fc19c002000]   320x240@[20.000000 20.000000]fps
[avfoundation @ 0x7fc19c002000]   320x240@[15.000015 15.000015]fps
[avfoundation @ 0x7fc19c002000]   320x240@[10.000000 10.000000]fps
[avfoundation @ 0x7fc19c002000]   320x240@[7.500002 7.500002]fps
[avfoundation @ 0x7fc19c002000]   320x240@[5.000000 5.000000]fps
[avfoundation @ 0x7fc19c002000]   352x288@[30.000030 30.000030]fps
[avfoundation @ 0x7fc19c002000]   352x288@[24.000038 24.000038]fps
[avfoundation @ 0x7fc19c002000]   352x288@[20.000000 20.000000]fps
[avfoundation @ 0x7fc19c002000]   352x288@[15.000015 15.000015]fps
[avfoundation @ 0x7fc19c002000]   352x288@[10.000000 10.000000]fps
[avfoundation @ 0x7fc19c002000]   352x288@[7.500002 7.500002]fps
[avfoundation @ 0x7fc19c002000]   352x288@[5.000000 5.000000]fps
[avfoundation @ 0x7fc19c002000]   432x240@[30.000030 30.000030]fps
[avfoundation @ 0x7fc19c002000]   432x240@[24.000038 24.000038]fps
[avfoundation @ 0x7fc19c002000]   432x240@[20.000000 20.000000]fps
[avfoundation @ 0x7fc19c002000]   432x240@[15.000015 15.000015]fps
[avfoundation @ 0x7fc19c002000]   432x240@[10.000000 10.000000]fps
[avfoundation @ 0x7fc19c002000]   432x240@[7.500002 7.500002]fps
[avfoundation @ 0x7fc19c002000]   432x240@[5.000000 5.000000]fps
[avfoundation @ 0x7fc19c002000]   640x480@[30.000030 30.000030]fps
[avfoundation @ 0x7fc19c002000]   640x480@[24.000038 24.000038]fps
[avfoundation @ 0x7fc19c002000]   640x480@[20.000000 20.000000]fps
[avfoundation @ 0x7fc19c002000]   640x480@[15.000015 15.000015]fps
[avfoundation @ 0x7fc19c002000]   640x480@[10.000000 10.000000]fps
[avfoundation @ 0x7fc19c002000]   640x480@[7.500002 7.500002]fps
[avfoundation @ 0x7fc19c002000]   640x480@[5.000000 5.000000]fps
[avfoundation @ 0x7fc19c002000]   800x448@[30.000030 30.000030]fps
[avfoundation @ 0x7fc19c002000]   800x448@[24.000038 24.000038]fps
[avfoundation @ 0x7fc19c002000]   800x448@[20.000000 20.000000]fps
[avfoundation @ 0x7fc19c002000]   800x448@[15.000015 15.000015]fps
[avfoundation @ 0x7fc19c002000]   800x448@[10.000000 10.000000]fps
[avfoundation @ 0x7fc19c002000]   800x448@[7.500002 7.500002]fps
[avfoundation @ 0x7fc19c002000]   800x448@[5.000000 5.000000]fps
[avfoundation @ 0x7fc19c002000]   864x480@[30.000030 30.000030]fps
[avfoundation @ 0x7fc19c002000]   864x480@[24.000038 24.000038]fps
[avfoundation @ 0x7fc19c002000]   864x480@[20.000000 20.000000]fps
[avfoundation @ 0x7fc19c002000]   864x480@[15.000015 15.000015]fps
[avfoundation @ 0x7fc19c002000]   864x480@[10.000000 10.000000]fps
[avfoundation @ 0x7fc19c002000]   864x480@[7.500002 7.500002]fps
[avfoundation @ 0x7fc19c002000]   864x480@[5.000000 5.000000]fps
[avfoundation @ 0x7fc19c002000]   800x600@[30.000030 30.000030]fps
[avfoundation @ 0x7fc19c002000]   800x600@[24.000038 24.000038]fps
[avfoundation @ 0x7fc19c002000]   800x600@[20.000000 20.000000]fps
[avfoundation @ 0x7fc19c002000]   800x600@[15.000015 15.000015]fps
[avfoundation @ 0x7fc19c002000]   800x600@[10.000000 10.000000]fps
[avfoundation @ 0x7fc19c002000]   800x600@[7.500002 7.500002]fps
[avfoundation @ 0x7fc19c002000]   800x600@[5.000000 5.000000]fps
[avfoundation @ 0x7fc19c002000]   1024x576@[30.000030 30.000030]fps
[avfoundation @ 0x7fc19c002000]   1024x576@[24.000038 24.000038]fps
[avfoundation @ 0x7fc19c002000]   1024x576@[20.000000 20.000000]fps
[avfoundation @ 0x7fc19c002000]   1024x576@[15.000015 15.000015]fps
[avfoundation @ 0x7fc19c002000]   1024x576@[10.000000 10.000000]fps
[avfoundation @ 0x7fc19c002000]   1024x576@[7.500002 7.500002]fps
[avfoundation @ 0x7fc19c002000]   1024x576@[5.000000 5.000000]fps
[avfoundation @ 0x7fc19c002000]   960x720@[30.000030 30.000030]fps
[avfoundation @ 0x7fc19c002000]   960x720@[24.000038 24.000038]fps
[avfoundation @ 0x7fc19c002000]   960x720@[20.000000 20.000000]fps
[avfoundation @ 0x7fc19c002000]   960x720@[15.000015 15.000015]fps
[avfoundation @ 0x7fc19c002000]   960x720@[10.000000 10.000000]fps
[avfoundation @ 0x7fc19c002000]   960x720@[7.500002 7.500002]fps
[avfoundation @ 0x7fc19c002000]   960x720@[5.000000 5.000000]fps
[avfoundation @ 0x7fc19c002000]   1280x720@[30.000030 30.000030]fps
[avfoundation @ 0x7fc19c002000]   1280x720@[24.000038 24.000038]fps
[avfoundation @ 0x7fc19c002000]   1280x720@[20.000000 20.000000]fps
[avfoundation @ 0x7fc19c002000]   1280x720@[15.000015 15.000015]fps
[avfoundation @ 0x7fc19c002000]   1280x720@[10.000000 10.000000]fps
[avfoundation @ 0x7fc19c002000]   1280x720@[7.500002 7.500002]fps
[avfoundation @ 0x7fc19c002000]   1280x720@[5.000000 5.000000]fps
[avfoundation @ 0x7fc19c002000]   1600x896@[30.000030 30.000030]fps
[avfoundation @ 0x7fc19c002000]   1600x896@[24.000038 24.000038]fps
[avfoundation @ 0x7fc19c002000]   1600x896@[20.000000 20.000000]fps
[avfoundation @ 0x7fc19c002000]   1600x896@[15.000015 15.000015]fps
[avfoundation @ 0x7fc19c002000]   1600x896@[10.000000 10.000000]fps
[avfoundation @ 0x7fc19c002000]   1600x896@[7.500002 7.500002]fps
[avfoundation @ 0x7fc19c002000]   1600x896@[5.000000 5.000000]fps
[avfoundation @ 0x7fc19c002000]   1920x1080@[30.000030 30.000030]fps
[avfoundation @ 0x7fc19c002000]   1920x1080@[24.000038 24.000038]fps
[avfoundation @ 0x7fc19c002000]   1920x1080@[20.000000 20.000000]fps
[avfoundation @ 0x7fc19c002000]   1920x1080@[15.000015 15.000015]fps
[avfoundation @ 0x7fc19c002000]   1920x1080@[10.000000 10.000000]fps
[avfoundation @ 0x7fc19c002000]   1920x1080@[7.500002 7.500002]fps
[avfoundation @ 0x7fc19c002000]   1920x1080@[5.000000 5.000000]fps
[avfoundation @ 0x7fc19c002000]   160x90@[30.000030 30.000030]fps
[avfoundation @ 0x7fc19c002000]   160x90@[24.000038 24.000038]fps
[avfoundation @ 0x7fc19c002000]   160x90@[20.000000 20.000000]fps
[avfoundation @ 0x7fc19c002000]   160x90@[15.000015 15.000015]fps
[avfoundation @ 0x7fc19c002000]   160x90@[10.000000 10.000000]fps
[avfoundation @ 0x7fc19c002000]   160x90@[7.500002 7.500002]fps
[avfoundation @ 0x7fc19c002000]   160x90@[5.000000 5.000000]fps
[avfoundation @ 0x7fc19c002000]   160x120@[30.000030 30.000030]fps
[avfoundation @ 0x7fc19c002000]   160x120@[24.000038 24.000038]fps
[avfoundation @ 0x7fc19c002000]   160x120@[20.000000 20.000000]fps
[avfoundation @ 0x7fc19c002000]   160x120@[15.000015 15.000015]fps
[avfoundation @ 0x7fc19c002000]   160x120@[10.000000 10.000000]fps
[avfoundation @ 0x7fc19c002000]   160x120@[7.500002 7.500002]fps
[avfoundation @ 0x7fc19c002000]   160x120@[5.000000 5.000000]fps
[avfoundation @ 0x7fc19c002000]   176x144@[30.000030 30.000030]fps
[avfoundation @ 0x7fc19c002000]   176x144@[24.000038 24.000038]fps
[avfoundation @ 0x7fc19c002000]   176x144@[20.000000 20.000000]fps
[avfoundation @ 0x7fc19c002000]   176x144@[15.000015 15.000015]fps
[avfoundation @ 0x7fc19c002000]   176x144@[10.000000 10.000000]fps
[avfoundation @ 0x7fc19c002000]   176x144@[7.500002 7.500002]fps
[avfoundation @ 0x7fc19c002000]   176x144@[5.000000 5.000000]fps
[avfoundation @ 0x7fc19c002000]   320x180@[30.000030 30.000030]fps
[avfoundation @ 0x7fc19c002000]   320x180@[24.000038 24.000038]fps
[avfoundation @ 0x7fc19c002000]   320x180@[20.000000 20.000000]fps
[avfoundation @ 0x7fc19c002000]   320x180@[15.000015 15.000015]fps
[avfoundation @ 0x7fc19c002000]   320x180@[10.000000 10.000000]fps
[avfoundation @ 0x7fc19c002000]   320x180@[7.500002 7.500002]fps
[avfoundation @ 0x7fc19c002000]   320x180@[5.000000 5.000000]fps
[avfoundation @ 0x7fc19c002000]   320x240@[30.000030 30.000030]fps
[avfoundation @ 0x7fc19c002000]   320x240@[24.000038 24.000038]fps
[avfoundation @ 0x7fc19c002000]   320x240@[20.000000 20.000000]fps
[avfoundation @ 0x7fc19c002000]   320x240@[15.000015 15.000015]fps
[avfoundation @ 0x7fc19c002000]   320x240@[10.000000 10.000000]fps
[avfoundation @ 0x7fc19c002000]   320x240@[7.500002 7.500002]fps
[avfoundation @ 0x7fc19c002000]   320x240@[5.000000 5.000000]fps
[avfoundation @ 0x7fc19c002000]   352x288@[30.000030 30.000030]fps
[avfoundation @ 0x7fc19c002000]   352x288@[24.000038 24.000038]fps
[avfoundation @ 0x7fc19c002000]   352x288@[20.000000 20.000000]fps
[avfoundation @ 0x7fc19c002000]   352x288@[15.000015 15.000015]fps
[avfoundation @ 0x7fc19c002000]   352x288@[10.000000 10.000000]fps
[avfoundation @ 0x7fc19c002000]   352x288@[7.500002 7.500002]fps
[avfoundation @ 0x7fc19c002000]   352x288@[5.000000 5.000000]fps
[avfoundation @ 0x7fc19c002000]   432x240@[30.000030 30.000030]fps
[avfoundation @ 0x7fc19c002000]   432x240@[24.000038 24.000038]fps
[avfoundation @ 0x7fc19c002000]   432x240@[20.000000 20.000000]fps
[avfoundation @ 0x7fc19c002000]   432x240@[15.000015 15.000015]fps
[avfoundation @ 0x7fc19c002000]   432x240@[10.000000 10.000000]fps
[avfoundation @ 0x7fc19c002000]   432x240@[7.500002 7.500002]fps
[avfoundation @ 0x7fc19c002000]   432x240@[5.000000 5.000000]fps
[avfoundation @ 0x7fc19c002000]   640x480@[30.000030 30.000030]fps
[avfoundation @ 0x7fc19c002000]   640x480@[24.000038 24.000038]fps
[avfoundation @ 0x7fc19c002000]   640x480@[20.000000 20.000000]fps
[avfoundation @ 0x7fc19c002000]   640x480@[15.000015 15.000015]fps
[avfoundation @ 0x7fc19c002000]   640x480@[10.000000 10.000000]fps
[avfoundation @ 0x7fc19c002000]   640x480@[7.500002 7.500002]fps
[avfoundation @ 0x7fc19c002000]   640x480@[5.000000 5.000000]fps
[avfoundation @ 0x7fc19c002000]   800x448@[30.000030 30.000030]fps
[avfoundation @ 0x7fc19c002000]   800x448@[24.000038 24.000038]fps
[avfoundation @ 0x7fc19c002000]   800x448@[20.000000 20.000000]fps
[avfoundation @ 0x7fc19c002000]   800x448@[15.000015 15.000015]fps
[avfoundation @ 0x7fc19c002000]   800x448@[10.000000 10.000000]fps
[avfoundation @ 0x7fc19c002000]   800x448@[7.500002 7.500002]fps
[avfoundation @ 0x7fc19c002000]   800x448@[5.000000 5.000000]fps
[avfoundation @ 0x7fc19c002000]   864x480@[24.000038 24.000038]fps
[avfoundation @ 0x7fc19c002000]   864x480@[20.000000 20.000000]fps
[avfoundation @ 0x7fc19c002000]   864x480@[15.000015 15.000015]fps
[avfoundation @ 0x7fc19c002000]   864x480@[10.000000 10.000000]fps
[avfoundation @ 0x7fc19c002000]   864x480@[7.500002 7.500002]fps
[avfoundation @ 0x7fc19c002000]   864x480@[5.000000 5.000000]fps
[avfoundation @ 0x7fc19c002000]   800x600@[24.000038 24.000038]fps
[avfoundation @ 0x7fc19c002000]   800x600@[20.000000 20.000000]fps
[avfoundation @ 0x7fc19c002000]   800x600@[15.000015 15.000015]fps
[avfoundation @ 0x7fc19c002000]   800x600@[10.000000 10.000000]fps
[avfoundation @ 0x7fc19c002000]   800x600@[7.500002 7.500002]fps
[avfoundation @ 0x7fc19c002000]   800x600@[5.000000 5.000000]fps
[avfoundation @ 0x7fc19c002000]   1024x576@[15.000015 15.000015]fps
[avfoundation @ 0x7fc19c002000]   1024x576@[10.000000 10.000000]fps
[avfoundation @ 0x7fc19c002000]   1024x576@[7.500002 7.500002]fps
[avfoundation @ 0x7fc19c002000]   1024x576@[5.000000 5.000000]fps
[avfoundation @ 0x7fc19c002000]   960x720@[15.000015 15.000015]fps
[avfoundation @ 0x7fc19c002000]   960x720@[10.000000 10.000000]fps
[avfoundation @ 0x7fc19c002000]   960x720@[7.500002 7.500002]fps
[avfoundation @ 0x7fc19c002000]   960x720@[5.000000 5.000000]fps
[avfoundation @ 0x7fc19c002000]   2304x1296@[2.000001 2.000001]fps
[avfoundation @ 0x7fc19c002000]   2304x1536@[2.000001 2.000001]fps
0: Input/output error
```

**Comment by almarklein** [comment]
It looks like ffmpeg does not accept the default fps or something. You could try explicitly specifying the fps with `imageio.get_reader('<video0>', fps=...)`. If needed you could also dive a bit deeper in the [related source code of the ffmpeg plugin](https://github.com/imageio/imageio/blob/master/imageio/plugins/ffmpeg.py#L464). It seems like the precision of the fps is limited to two decimals, not sure if that can be a problem ...

You can also try `<video1>` etc.  If you have multiple cameras.

**Comment by OrganicIrradiation** [comment]
I'm having the same issue with ffmpeg 4.0.2 on macOS 10.14.

Rather than ```imageio.get_reader('<video0>', fps=...)``` try ```imageio.get_reader('<video0>', input_params=['-framerate', '30'])```. Worked for me.

**Comment by almarklein** [comment]
Should be fixed per #510, please reopen if not.

**Comment by dennisvang** [comment]
Here's a recent ffmpeg ticket that may be related: [Unsupported frame rate with capture device after updating to MacOS Monterey](https://trac.ffmpeg.org/ticket/9505)
